### DISCUSSION TITLE
feature idea: FITdump

### DISCUSSION THREAD

**Discussion by RinseRepeat** [muktihari/fit#189] [Ideas]
mimicking Garmin's FitCSV is great work, but take a look at the tool found in the FIT.pm  (package Garmin::FIT) Perl library for FIT 

https://github.com/mrihtar/Garmin-FIT

**_FITdump.pl_**
https://github.com/mrihtar/Garmin-FIT/blob/master/fitdump.pl

The output it makes is excellent and even handles some undocumented Garmin data and devices

Sadly no longer developed so getting outdated but still works.

Makes it incredibly easy to search and debug data in FIT files, even more so than CSV which can be unreadable at times.

**Comment by muktihari** [comment]
Hi @RinseRepeat, thanks for sharing an idea.
What I can summarize from the link you shared, the perl's `FITdump` can convert FIT into GPX, SLF, JSON and TXT back and forth, isn't it?

For converting FIT, GPX and TCX back and forth, I created an app few months ago, it's 100% client-side, https://openivity.github.io, you can view and convert those files in interactive UI. Maybe this is what you looking for (?)

I believe SLF format is not commonly used nowadays (I personally don't use it and don't have any files in that format). Supporting it is probably out of scope of what this repository aims to achieve. However, we can set up new repository for that purpose as long as you can convince me that this SLF format encoding is really useful and of course, only when I have spare time to do it since I need to learn how the encoding works.

JSON format is a common file format and supporting it seems reasonable to me, though what data to be put there is still a bit ambiguous, just like the TXT format. What data do you want to view (that the current CSV file is not enough or hard to read)? Can you give me examples of files generated by the `FITdump`? I have no perl installed in my machine so couldn't try it, having the example of the files will be helpful.

> The output it makes is excellent and even handles some undocumented Garmin data and devices

What kind of undocumented Garmin data and devices that you want to handle? More context about this will be helpful.

> Makes it incredibly easy to search and debug data in FIT files, even more so than CSV which can be unreadable at times.

Having the examples of `CSV can be unredable at a times` will also be useful. I need example of real world use cases so I could understand what we can do about it.

**Reply by RinseRepeat** [reply]
FITdump is just raw plain text file of all data inside a FIT file, useful for debugging

I am attaching an example, easier to show than explain.

(I've removed serial numbers and other personal info, there's no GPS data as indoor ride)

The functionality is fantastic, it's just years out of date and not updated anymore.

[E1O70718.TXT](https://github.com/muktihari/fit/files/15013429/E1O70718.TXT)

**Reply by muktihari** [reply]
Thanks for providing the example file, the output is similar to what `fitprint` was printing, I just update it to print more details. Try this: [https://github.com/muktihari/fit/tree/master/cmd/fitprint](https://github.com/muktihari/fit/tree/master/cmd/fitprint).
